app/tests.py

```diff
@@ -0,0 +1,137 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from app.models import Recipe, Ingredient
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from app.serializers import RecipeSerializer
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def create_recipe(**params):
+    defaults = {
+        'name': 'Sample name',
+        'description': 'Sample description'
+    }
+    defaults.update(params)
+
+    recipe = Recipe.objects.create(**defaults)
+
+    return recipe
+
+
+def detail_url(recipe_id):
+    """Create and return a recipe detail URL."""
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+# Create your tests here.
```

Comment:
With files that we don't leave as-is, we should delete autogenerated comments

Reply:
Ok, removed.